### PR TITLE
refactor(error-handling): add exposed component + route-level error boundaries

### PR DESCRIPTION
**Pull request by shelldandy**
## Summary

**Related issue:** https://github.com/grafana/metrics-drilldown/issues/1252

After #1244 replaced the global `useCatchExceptions` hook with a scoped `<AppErrorBoundary>` at the app root, three gaps remained: exposed components had no error handling at all, scene crashes took down the entire page, and the deleted hook left a no-op shim behind. This PR closes all three.

## What changed

**Exposed component error boundaries.** `EntityMetrics`, `LabelBreakdown`, `MiniBreakdown`, and `SourceMetrics` mount outside the app tree in host Grafana surfaces (RCA Workbench, Grafana Assistant). Each `Lazy*.tsx` wrapper now wraps its `<Suspense>` in a shared `<ExposedComponentErrorBoundary>` that catches chunk-load failures and runtime render errors, renders an inline `<Alert severity="error">` (not the full-page `ErrorView`), and logs through the isolated Faro instance. The logger import is dynamic to keep it out of the synchronous entry chunk.

**Route-level error boundary.** `TrailErrorBoundary` wraps `<trail.Component>` inside `Trail.tsx`, so scene crashes render `ErrorView` while preserving `PluginPage` chrome (breadcrumbs, header). It passes `dependencies={[pathname]}` so navigating away via breadcrumbs resets the boundary and recovers. `ChunkLoadError` is not handled here; it propagates to `AppErrorBoundary` for soft-reload recovery.

**Deleted `useCatchExceptions` shim.** The no-op shim at `src/App/useCatchExceptions.ts` had no remaining consumers. Removed.

**Spelling fix.** Corrected `handheldBy` → `handledBy` across all error-handling logger context (including pre-existing code in `AppErrorBoundary`, `ErrorView`, and `displayStatus`).

### Error boundary layering

```mermaid
graph TB
    A["AppErrorBoundary<br/>(app root)"] --> B["ChunkLoadError → soft-reload"]
    A --> C["Other errors → ErrorView"]
    A --> D["TrailErrorBoundary<br/>(route level, resets on pathname change)"]
    D --> E["Scene errors → ErrorView<br/>(PluginPage chrome preserved)"]
    F["ExposedComponentErrorBoundary<br/>(per exposed component)"] --> G["Inline Alert<br/>(no full-page takeover)"]
```

## Tests

17 new tests across 2 suites:

- **`ExposedComponentErrorBoundary.test.tsx`** (6): happy path, error display, dynamic-import logger context, component name interpolation, edge cases.
- **`TrailErrorBoundary.test.tsx`** (6 + 1 new recovery test): happy path, ErrorView rendering, logger context, chunk error passthrough, stack trace visibility, and navigation-recovery (mount error → change URL → assert children re-render).

---

[Compound Engineering](https://github.com/EveryInc/compound-engineering-plugin)
Pi

### CODE REVIEW

File: src/App/TrailErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * Sits inside PluginPage so the page chrome (breadcrumbs, header) remains
+ * visible when a scene crashes. The user can still navigate away via
+ * breadcrumbs to recover.
+ *
+ * ChunkLoadError recovery is intentionally NOT handled here — those errors
+ * propagate to AppErrorBoundary at the app root for soft-reload handling.
+ */
+export function TrailErrorBoundary({ children }: Readonly<{ children: React.ReactNode }>) {
+  return (
+    <ErrorBoundary boundaryName="metrics-drilldown-trail" errorLogger={errorLogger}>
```
Comment (bohandley) [inline src/App/TrailErrorBoundary.tsx]: In the PR description it say "Users can navigate away via breadcrumbs to recover," but I think we need to pass dependencies here with the path. `AppErrorBoundary` does this but TrailErrorBoundary is missing this.

```
const { pathname } = useLocation();
...
<ErrorBoundary boundaryName="metrics-drilldown-app" errorLogger={errorLogger} dependencies={[pathname]}>
```

Can we add a test that mounts an error, changes the URL through the router, and asserts the children re-render? The current "renders ErrorView when child throws" test (`src/App/TrailErrorBoundary.test.tsx:43-51`) does not exercise recovery.

File: src/exposedComponents/ExposedComponentErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  return (
+    <ErrorBoundary boundaryName={boundaryName} errorLogger={errorLoggerFor(componentName)}>
+      {({ error }) => {
+        if (error) {
+          return (
+            <Alert
+              severity="error"
+              title={t('exposed-component-error-boundary.title', 'Something went wrong')}
+            >
+              {t(
```
Comment (bohandley) [inline src/exposedComponents/ExposedComponentErrorBoundary.tsx]: `errorLoggerFor(componentName)` is called inline in JSX so a new function reference is produced every render. This is an easy fix to memo-ize this :)

File: src/App/TrailErrorBoundary.test.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+  it('logs error through our isolated Faro instance with trail-error-boundary context', () => {
+    renderWithRouter(
+      <TrailErrorBoundary>
+        <ThrowingChild error={new Error('scene crash')} />
+      </TrailErrorBoundary>
+    );
+
+    expect(mockLogger.error).toHaveBeenCalledWith(expect.objectContaining({ message: 'scene crash' }), {
+      handheldBy: 'trail-error-boundary',
```
Comment (bohandley) [inline src/App/TrailErrorBoundary.test.tsx]: Spelling error I believe. This spelling is pretty much all through the PR.

handheldBy => handledBy

**Review by bohandley — CHANGES_REQUESTED** [review]
Great work!!! The PR is a clean, focused completion of the boundary refactor started in #1244. The three-layer model (AppErrorBoundary at the root, TrailErrorBoundary scoped inside PluginPage, ExposedComponentErrorBoundary per exposed surface) is correct. One functional concern may invalidate a claim in the PR body, that I'm requesting changes for. The others, memo-ize and spelling are nits :)

Thank you @shelldandy!!!

File: src/exposedComponents/ExposedComponentErrorBoundary.tsx (hunk 2 above)
Comment (shelldandy) [reply src/exposedComponents/ExposedComponentErrorBoundary.tsx]: Done — wrapped in `useMemo(() => errorLoggerFor(componentName), [componentName])` so the reference is stable across renders. → a8673aa

**Review by bohandley — APPROVED** [review]
LGTM! Thank you for the changes 🚀
